### PR TITLE
Add a new Symfony securty guard

### PR DESCRIPTION
**Pull request by jonathangreen**
## GitHub Issue

Part of:
https://github.com/Islandora-CLAW/CLAW/issues/565

## What does this Pull Request do?

A new guard that we can use in Silex that should have the same behavior
as the java Syn code, so we can secure our microservices in the same way
and use the same tokens everywhere.

It also adds a new ServiceProvider that will setup all the Islandora services for an app.

## How should this be tested?

There are a bunch of unit tests with this, but to test we will need the pull request to Crayfish that will actually use this component. 

Pull request to use this is here:
https://github.com/Islandora-CLAW/Crayfish/pull/24

## Interested parties
@Islandora-CLAW/committers

### CODE REVIEW

**Comment by jonathangreen** [comment]
Don't think this will ever go fully green because the master is missing codecov, travis is happy though.

File: src/IslandoraServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        // Only use logger if configured
+        if (strtolower($config['loglevel']) === 'none') {
+            $app['monolog'] = function () {
+                return new Logger('null', [new NullHandler()]);
+            };
+        } else {
+            $app->register(new MonologServiceProvider(), [
+                'monolog.logfile' => $config['logfile'],
+                'monolog.level' => $config['loglevel'],
+                'monolog.name' => 'Houdini',
```
Comment (whikloj) [inline src/IslandoraServiceProvider.php]: Leftovers from copy/paste?

**Comment by whikloj** [comment]
@jonathangreen this needs a rebase please.

**Comment by jonathangreen** [comment]
Rebased. Should be good to go.

**Pull request merged**